analyzers/src/SonarAnalyzer.CSharp/Helpers/InvocationExpressionSyntaxExtensions.cs

```diff
@@ -31,6 +32,11 @@ internal static class InvocationExpressionSyntaxExtensions
             memberAccessExpressionSyntax.NameIs(identifierName) &&
             semanticModel.GetTypeInfo(invocation).Type.Is(knownType);
 
+        internal static bool IsMemberAccessOnKnownType(this InvocationExpressionSyntax invocation, string identifierName, ImmutableArray<KnownType> knownTypes, SemanticModel semanticModel) =>
+            invocation.Expression is MemberAccessExpressionSyntax memberAccessExpressionSyntax &&
+            memberAccessExpressionSyntax.NameIs(identifierName) &&
+            semanticModel.GetSymbolInfo(invocation).Symbol.ContainingType.IsAny(knownTypes);
```

Comment:
Possible NRE, as discovered in the `IsIVMemberAccess` comment.

I'm not sure how `.Type` behaves in the method above. Probably the same?
```suggestion
            semanticModel.GetSymbolInfo(invocation).Symbol?.ContainingType.IsAny(knownTypes);
```